src/uwtools/config.py

```diff
@@ -80,6 +80,36 @@ def dump_file(self, output_path):
         ''' Interface to write a config object to a file at the
         output_path provided. '''
 
+    def compare_config(self, user_dict, base_dict=None):
+        #pylint: disable=multiple-statements
+        '''
+        Assuming a section, key/value structure of configuration types,
+        compare the dictionary to the values stored in the external file.
+        '''
+        if base_dict is None:
+            base_dict = self.data
+
+        #Set up lists for missing headers, keys and values between dicts
+        misheader = {}
+        miskey = {}
+        misvalue = {}
+
+        if len(base_dict.items()) != len(user_dict.items()):
+            raise KeyError('Given dictionaries have different ranges of keys!')
```

Comment:
I think that we'd want to know what those differences are. It's helpful to know if one config file contains a section that another does not. I think it may be useful to got through dict1, then dict2 to assess what exists/doesn't in each. Also _how_ they differ, I think is important.

I can make a couple small mods while you're away to capture that.

Reply:
Fair, and thanks! I had initially set it to just reject from the spec, but just giving the differences regardless is a better way for what we want to so